tests/dynamic_data/base_test.py

```diff
@@ -1,29 +1,33 @@
 import copy
+import os
 import pytest
 from test_tools import process_client, cleanup
-from .. import Publication
+from .. import Publication, TestTypes, ParametrizationKeys
 
 
 def pytest_generate_tests(metafunc):
     # used for parametrizing subclasses of TestSingleRestPublication, called once per each test function
     # https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-generate-tests
+    test_type_str = os.getenv(ParametrizationKeys.TEST_TYPE.value, TestTypes.MANDATORY.value)
+    test_type = TestTypes(test_type_str)
     cls = metafunc.cls
     rest_methods = cls.rest_parametrization['method']
     argvalues = []
     ids = []
     for key, params in cls.test_cases.items():
-        for method_function_name, method_name in rest_methods.items():
-            publ_name = cls.key_to_publication_base_name(key) + f"_{method_name}"
-            publication = Publication(cls.workspace, cls.publication_type, publ_name)
-            rest_method = getattr(cls, method_function_name)
-            argvalues.append([
-                publication,
-                key,
-                copy.deepcopy(params),
-                rest_method,
-                (publication, method_name),
-            ])
-            ids.append(publ_name)
+        if test_type == TestTypes.ALL or params.get(ParametrizationKeys.TEST_TYPE, cls.default_test_type) == TestTypes.MANDATORY:
```

Comment:
Maybe filter cls.test_cases before for cycle, and then use filtered dictionary for for cycle.